back/app/services/timeline_service.rb

```diff
@@ -33,17 +33,14 @@ def phase_is_complete?(phase, time = Time.now)
     phase.end_at.present? && phase.end_at <= date
   end
 
-  def current_or_last_can_contain_ideas_phase(project, time = Time.now)
```

Comment:
@kogre I renamed this as I didn't like the "can_contain_ideas" part in the name. This returns the phase that is "the most related" to the transitive (ideation) custom form. I hope one day we can also reuse this in other parts (I was thinking e.g. determining if commenting/reacting is possible, but that logic is probably more related to the input we're looking at and which phases it's linked to).

The line:
```
return current if current_method&.transitive? || (current_method&.supports_posting_inputs? && !current_method&.never_show?) # ideation, voting, proposals, but not native surveys
```
was the most challenging part.

Reply:
It can also be useful to look at the corresponding specs (at the bottom of this PR) to understand the behaviour better, which I've updated.

Reply:
Would be good to add the explanation of what it does as a comment to the code too. I'm thinking a bit more about this one, feels like it's a source of complexity we might not (always) really need.

Reply:
It would be great if we could find a concept that is more intuitive. I'll add a comment for now.